File: guides/building-a-rest-api-spring-boot-vs-micronaut-security-basic-auth/micronautframework/java/src/main/java/example/micronaut/SaasSubscriptionController.java
/*
 * Copyright 2017-2024 original authors
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
package example.micronaut;

import io.micronaut.http.HttpResponse;
import io.micronaut.http.annotation.Controller;
import io.micronaut.http.annotation.Get;
import io.micronaut.http.annotation.PathVariable;
import java.security.Principal;
import io.micronaut.security.annotation.Secured;

@Controller("/subscriptions") // <1>
@Secured("SAAS_SUBSCRIPTION_OWNER") // <2>
class SaasSubscriptionController {

    private final SaasSubscriptionRepository repository;

    SaasSubscriptionController(SaasSubscriptionRepository repository) { // <3>
        this.repository = repository;
    }

    @Get("/{id}") // <4>
    HttpResponse<SaasSubscription> findById(@PathVariable Long id, // <5>
                                            Principal principal) { // <6>
        return repository.findByIdAndOwner(id, principal.getName())
                .map(HttpResponse::ok)
                .orElseGet(HttpResponse::notFound); // <7>
    }
}

File: guides/building-a-rest-api-spring-boot-vs-micronaut-get-list/micronautframework/kotlin/src/main/kotlin/example/micronaut/SaasSubscriptionController.kt
/*
 * Copyright 2017-2026 original authors
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
package example.micronaut

import io.micronaut.http.HttpResponse
import io.micronaut.http.annotation.Controller
import io.micronaut.http.annotation.Get
import io.micronaut.http.annotation.PathVariable

@Controller("/subscriptions") // <1>
class SaasSubscriptionController(private val repository: SaasSubscriptionRepository) {

    @Get("/{id}") // <2>
    fun findById(@PathVariable id: Long): HttpResponse<SaasSubscription> = // <3>
        repository.findById(id)
            .map { HttpResponse.ok(it) }
            .orElseGet { HttpResponse.notFound() } // <4>
}
